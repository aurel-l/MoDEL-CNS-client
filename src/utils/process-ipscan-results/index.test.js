import processIPScanResults from '.';

describe('processIPScanResults', () => {
  it('should return empty array when no match passed', () => {
    expect(processIPScanResults()).toEqual([]);
    expect(processIPScanResults(null)).toEqual([]);
    expect(processIPScanResults([])).toEqual([]);
  });

  it('should return a transformed array of matches', () => {
    expect(processIPScanResults(require('./example1.json'))).toMatchSnapshot();
    expect(processIPScanResults(require('./example2.json'))).toMatchSnapshot();
  });
});
