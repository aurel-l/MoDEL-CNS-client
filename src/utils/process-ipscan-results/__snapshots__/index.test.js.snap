// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processIPScanResults should return a transformed array of matches 1`] = `
Array [
  Object {
    "accession": "IPR001019",
    "color": "hsla(178, 58%, 53%, 1)",
    "db": "InterPro",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 51,
            "start": 1,
          },
        ],
      },
    ],
    "signatures": Array [
      Object {
        "accession": "PTHR10218",
        "color": "hsla(178, 58%, 53%, 0.25)",
        "db": "PANTHER",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 51,
                "start": 1,
              },
            ],
          },
        ],
      },
    ],
  },
  Object {
    "accession": "IPR027417",
    "color": "hsla(128, 48%, 53%, 1)",
    "db": "InterPro",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 51,
            "start": 15,
          },
        ],
      },
    ],
    "signatures": Array [
      Object {
        "accession": "SSF52540",
        "color": "hsla(128, 48%, 53%, 0.25)",
        "db": "SUPERFAMILY",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 51,
                "start": 15,
              },
            ],
          },
        ],
      },
    ],
  },
  Object {
    "accession": "mobidb-lite",
    "color": "hsla(196, 56%, 51%, 1)",
    "db": "MOBIDB_LITE",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 25,
            "start": 1,
          },
        ],
      },
    ],
  },
  Object {
    "accession": "Coil",
    "color": "hsla(31, 51%, 46%, 1)",
    "db": "COILS",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 21,
            "start": 1,
          },
        ],
      },
    ],
  },
  Object {
    "accession": "PTHR10218:SF306",
    "color": "hsla(216, 56%, 51%, 1)",
    "db": "PANTHER",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 51,
            "start": 1,
          },
        ],
      },
    ],
  },
  Object {
    "accession": "G3DSA:3.40.50.300",
    "color": "hsla(324, 44%, 49%, 1)",
    "db": "GENE3D",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 51,
            "start": 2,
          },
        ],
      },
    ],
  },
]
`;

exports[`processIPScanResults should return a transformed array of matches 2`] = `
Array [
  Object {
    "accession": "IPR001019",
    "color": "hsla(178, 58%, 53%, 1)",
    "db": "InterPro",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 298,
            "start": 1,
          },
        ],
      },
    ],
    "signatures": Array [
      Object {
        "accession": "cd00066",
        "color": "hsla(178, 58%, 53%, 0.25)",
        "db": "CDD",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 292,
                "start": 4,
              },
            ],
          },
        ],
      },
      Object {
        "accession": "PR00318",
        "color": "hsla(178, 58%, 53%, 0.25)",
        "db": "PRINTS",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 123,
                "start": 101,
              },
            ],
          },
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 200,
                "start": 191,
              },
            ],
          },
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 147,
                "start": 130,
              },
            ],
          },
        ],
      },
      Object {
        "accession": "PTHR10218",
        "color": "hsla(178, 58%, 53%, 0.25)",
        "db": "PANTHER",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 298,
                "start": 1,
              },
            ],
          },
        ],
      },
      Object {
        "accession": "SM00275",
        "color": "hsla(178, 58%, 53%, 0.25)",
        "db": "SMART",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 297,
                "start": 1,
              },
            ],
          },
        ],
      },
      Object {
        "accession": "PF00503",
        "color": "hsla(178, 58%, 53%, 0.25)",
        "db": "PFAM",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 287,
                "start": 6,
              },
            ],
          },
        ],
      },
    ],
  },
  Object {
    "accession": "IPR011025",
    "color": "hsla(326, 46%, 51%, 1)",
    "db": "InterPro",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 115,
            "start": 2,
          },
        ],
      },
    ],
    "signatures": Array [
      Object {
        "accession": "G3DSA:1.10.400.10",
        "color": "hsla(326, 46%, 51%, 0.25)",
        "db": "GENE3D",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 115,
                "start": 2,
              },
            ],
          },
        ],
      },
      Object {
        "accession": "SSF47895",
        "color": "hsla(326, 46%, 51%, 0.25)",
        "db": "SUPERFAMILY",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 114,
                "start": 4,
              },
            ],
          },
        ],
      },
    ],
  },
  Object {
    "accession": "IPR000367",
    "color": "hsla(63, 43%, 48%, 1)",
    "db": "InterPro",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 29,
            "start": 14,
          },
        ],
      },
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 44,
            "start": 30,
          },
        ],
      },
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 189,
            "start": 174,
          },
        ],
      },
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 252,
            "start": 238,
          },
        ],
      },
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 266,
            "start": 253,
          },
        ],
      },
    ],
    "signatures": Array [
      Object {
        "accession": "PR00443",
        "color": "hsla(63, 43%, 48%, 0.25)",
        "db": "PRINTS",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 29,
                "start": 14,
              },
            ],
          },
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 252,
                "start": 238,
              },
            ],
          },
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 44,
                "start": 30,
              },
            ],
          },
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 266,
                "start": 253,
              },
            ],
          },
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 189,
                "start": 174,
              },
            ],
          },
        ],
      },
    ],
  },
  Object {
    "accession": "IPR027417",
    "color": "hsla(128, 48%, 53%, 1)",
    "db": "InterPro",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 296,
            "start": 117,
          },
        ],
      },
    ],
    "signatures": Array [
      Object {
        "accession": "SSF52540",
        "color": "hsla(128, 48%, 53%, 0.25)",
        "db": "SUPERFAMILY",
        "locations": Array [
          Object {
            "fragments": Array [
              Object {
                "dc-status": "CONTINUOUS",
                "end": 296,
                "start": 117,
              },
            ],
          },
        ],
      },
    ],
  },
  Object {
    "accession": "PTHR10218:SF320",
    "color": "hsla(160, 40%, 45%, 1)",
    "db": "PANTHER",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 298,
            "start": 1,
          },
        ],
      },
    ],
  },
  Object {
    "accession": "Coil",
    "color": "hsla(31, 51%, 46%, 1)",
    "db": "COILS",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 25,
            "start": 5,
          },
        ],
      },
    ],
  },
  Object {
    "accession": "G3DSA:3.40.50.300",
    "color": "hsla(79, 59%, 54%, 1)",
    "db": "GENE3D",
    "locations": Array [
      Object {
        "fragments": Array [
          Object {
            "dc-status": "CONTINUOUS",
            "end": 287,
            "start": 116,
          },
        ],
      },
    ],
  },
]
`;
